Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Auth.hs

```diff
@@ -79,40 +80,41 @@ verifyAdmin user = do
     throwError AccessDenied
   return user
 
-verifyToken :: (HasEsqEnv m r, MonadFlow m) => RegToken -> m SR.RegistrationToken
+verifyToken :: (HasEsqEnv m r, MonadFlow m, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m SR.RegistrationToken
 verifyToken regToken = do
   QR.findByToken regToken
     >>= Utils.fromMaybeM (InvalidToken regToken)
     >>= validateToken
 
-validateAdmin :: (HasEsqEnv m r, EncFlow m r) => RegToken -> m Person.Person
+validateAdmin :: (HasEsqEnv m r, EncFlow m r, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m Person.Person
 validateAdmin regToken = do
   SR.RegistrationToken {..} <- verifyToken regToken
   user <-
     QP.findById (Id entityId)
       >>= fromMaybeM (PersonNotFound entityId)
   verifyAdmin user
 
-verifyPerson :: (HasEsqEnv m r, Redis.HedisFlow m r, HasField "authTokenCacheExpiry" r Seconds, MonadFlow m) => RegToken -> m (Id Person.Person, Id Merchant.Merchant)
+verifyPerson :: (HasEsqEnv m r, Redis.HedisFlow m r, HasField "authTokenCacheExpiry" r Seconds, MonadFlow m, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m (Id Person.Person, Id Merchant.Merchant, Id DMOC.MerchantOperatingCity)
 verifyPerson token = do
   let key = authTokenCacheKey token
   authTokenCacheExpiry <- getSeconds <$> asks (.authTokenCacheExpiry)
   result <- Redis.safeGet key
```

Comment:
For older token, it will give not case match `(personId, merchantId, merchantOperatingCityId)` hence will break. 
So do like this
```
  case result of
    Just (personId, merchantId) -> do
      merchantOperatingCityId <- defaultCityForMerchantId
      return (personId, merchantId, merchantOperatingCityId)
    Just (personId, merchantId, merchantOperatingCityId) -> return (personId, merchantId, merchantOperatingCityId)
    Nothing -> do
      sr <- verifyToken token
      let expiryTime = min sr.tokenExpiry authTokenCacheExpiry
      let personId = Id sr.entityId
      let merchantId = Id sr.merchantId
      Redis.setExp key (personId, merchantId) expiryTime
      return (personId, merchantId)
      let merchantOperatingCityId = Id sr.merchantOperatingCityId
      Redis.setExp key (personId, merchantId, merchantOperatingCityId) expiryTime
      return (personId, merchantId, merchantOperatingCityId)
 ```
 
 Also handle the case of `sr.merchantOperatingCityId` it will be Maybe as added comment in its table file.

Reply:
Used the following type:
Maybe (Id Person, Id Merchant, Maybe (Id MerchantOperatingCity))